src/cluster_asm.c

```diff
@@ -668,17 +672,11 @@ void clusterSyncSlotsSnapshotEOF(client *c) {
         "RDB channel snapshot transfer done for task");
 
     task->state = ASM_REPLAY_STREAM;
-  
-    client *main_channel_client = createClient(task->main_channel_conn);
-    main_channel_client->flags |= CLIENT_MASTER;
-    main_channel_client->querybuf = sdsempty();
-    main_channel_client->authenticated = 1;
-    main_channel_client->user = NULL;
-    main_channel_client->task = task;
 
     /* Replay stream. */
-    // TODO: replay the buffered stream from the main channel connection.
+    asmSyncBufferStreamToDb(task);
     serverLog(LL_NOTICE, "Replaying stream for task is done");
+    connSetReadHandler(task->main_channel_conn, readQueryFromClient);
```

Comment:
why do we need this? We already received EOF at this point right?

Reply:
here, the EOF we received is snapshot EOF, we need to continue reading stream EOF

Reply:
ah ok, I was confused. Okay, so at this point, we replayed all the accumulated buffer. Now, we like to read until we receive EOF.